atlp-rwanda/script-heroes-bn-backend
# #174797504 setup sequelize

#### What does this PR do?
configure sequelize to be used as our ORM on the API.
#### Description of Task to be completed?
- install Sequelize and Sequelize-cli
- setup .sequeliceC file
- configure squelize
- Set up the migrations and models structure
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout the branch `git checkout ch-setup-sequelize-#174797509`
- Create a .squelizeC file in the project root directory
- Configure the database/config.js file and fill out the required dtata
- Creating first Model (and Migration) with `sequelize model:generate --name User --attributes firstName:string,lastName:string,email:string`
- Running Migrations with `sequelize db:migrate`
- Creating First Seed with `sequelize seed:generate --name demo-user`
- Running Seeds with `sequelize db:seed:all `
- Switch to your database with this command `psql your_database_name`
- Run this query to make sure your data is in database `SELECT * FROM "Users"";`
#### What are the relevant pivotal tracker stories?
[#174797504](https://www.pivotaltracker.com/story/show/174797504)
